src/screens/WalletScreen/WalletScreen.tsx

```diff
@@ -82,6 +91,37 @@ function WalletScreen() {
     });
   }, []);
 
+  // We cannot rely on `onMomentumScrollEnd` because it's not called when the user scrolls directly rather than swiping
+  const debouncedAddSubscribedTokensRef = useRef(
+    debounce((viewableItems, routeName) => {
+      const viewableTokenUniqueIds = extractTokenRowIds(viewableItems);
+      if (viewableTokenUniqueIds.length > 0) {
+        addSubscribedTokens({ route: routeName, tokenIds: viewableTokenUniqueIds });
+      }
+    }, 250)
+  );
```

Comment:
Would be good to use `useStableValue` or `useLazyRef` here so `debounce` isn't invoked every render.